### DISCUSSION TITLE
Pagination Exceeded Exception

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#37052] [Ideas]
In a current inertia project the query string is updated with filters that may reduce the results below current page. Like search performed on page 5

But since the ?page=5 is in the query string, the results are empty.

I used the following exception to redirect to `?page=1` in that scenario.

```php
$results = Post::paginate()->withQueryString();

if(request()->input('page') > $results->lastPage()) {
    throw new PageExceededException();
}
```
```php
<?php

namespace App\Exceptions;

use Exception;

class PageExceededException extends Exception
{
    /**
     * The http status code.
     * @var int
     */
    protected $code = 302;

    /**
     * Render the exception into an HTTP response.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return \Illuminate\Http\RedirectResponse
     */
    public function render($request)
    {
        return redirect()->to(request()->url() . '?' . http_build_query(
            array_merge($request->query(), [ 'page' => 1 ])
        ));
    }
}

```
✅ Passing test
```php
public function test_redirect_page_if_exceeds_total_pages()
{
    $this->get('/posts?page=2&search=hello')
        ->assertRedirect('/posts?page=1&search=hello');
}
```
Sharing incase someone wants to PR it as I don't know how that would fit into the framework.

**Comment by selcukcukur** [comment]
This is a situation that can be set by the developer within the application with different methods. I think there is no need to add it to the frame

**Reply by dillingham** [reply]
@selcukcukur I mimicked the behavior of form request validations, which throw an exception when rules aren't met and it redirects vs renders a page for the error code. So one could make the point that a pagination page exceeding it's total would be invalid or a validation error and also in need of redirect. But its lasted this long without so maybe you're right :)
